client/lib/url/test/is-external.js

```diff
@@ -0,0 +1,143 @@
+		const actual = isExternal( urlWithoutHttp );
+		expect( actual ).toBeTruthy();
+	} );
+
+	test( 'should return true for url without http and path', () => {
+		const urlWithoutHttp = 'en.support.wordpress.com/start';
+		const actual = isExternal( urlWithoutHttp );
+		expect( actual ).toBeTruthy();
+	} );
+
+	test( 'should return true for url without http and // path', () => {
+		const urlWithoutHttp = 'en.support.wordpress.com//start';
+		const actual = isExternal( urlWithoutHttp );
+		expect( actual ).toBeTruthy();
+	} );
+
+	test( 'should return true for just external relative // path', () => {
+		const urlWithoutHttp = '//manage';
+		const actual = isExternal( urlWithoutHttp );
+		expect( actual ).toBeTruthy();
+	} );
+
+	describe( 'with global.window (test against window.location.hostname)', () => {
+		// window.location.hostname === 'example.com'
+		test( 'should return false for for internal protocol-relative url', () => {
+			const source = '//example.com';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeFalsy();
+		} );
+
+		test( 'should return true for for external protocol-relative url', () => {
+			const source = '//some-other-site.com';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeTruthy();
+		} );
+
+		test( 'should return false for internal url', () => {
+			const source = 'https://example.com';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeFalsy();
+		} );
+
+		test( 'should return true for external url', () => {
+			const source = 'https://not-example.com';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeTruthy();
+		} );
+
+		test( 'should return false for internal path without http', () => {
+			const urlWithoutHttp = 'example.com//me';
+			const actual = isExternal( urlWithoutHttp );
+			expect( actual ).toBeFalsy();
+		} );
+
+		test( 'should return true for internal but legacy path with http', () => {
+			const urlWithoutHttp = 'http://example.com/manage';
+			const actual = isExternal( urlWithoutHttp );
+			expect( actual ).toBeTruthy();
+		} );
+
+		test( 'should return true for internal but legacy path without http', () => {
+			const urlWithoutHttp = 'example.com/manage';
+			const actual = isExternal( urlWithoutHttp );
+			expect( actual ).toBeTruthy();
+		} );
+
+		test( 'should return true for subdomains', () => {
+			const source = '//ns1.example.com';
+			const actual = isExternal( source );
+			expect( actual ).toBeTruthy();
+		} );
+	} );
+
+	describe( 'without global.window (test against config hostname)', () => {
+		test( 'should return false for internal protocol-relative url', () => {
+			const source = '//calypso.localhost';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeFalsy();
+		} );
+
+		test( 'should return true for external protocol-relative url', () => {
+			const source = '//some-other-site.com';
+
+			const actual = isExternal( source );
+
+			expect( actual ).toBeTruthy();
```

Comment:
When we want exactly `true`, we should prefer this form:

```suggestion
			expect( actual ).toBe( true );
```

Not a big deal 🙂

Reply:
Yep, we can be stricter 🙂 Changed everything to `.toBe( true )`/`.toBe( false )`.